services/spar/src/Spar/App.hs

```diff
@@ -58,11 +60,20 @@ instance HasConfig Spar where
 
 instance SP Spar where
   -- FUTUREWORK: optionally use 'field' to index user or idp ids for easier logfile processing.
-  -- (FUTUREWORK: we could also call 'show' on the message instead of manually removing newlines,
-  -- then we could cut&paste it into ghci and make everything visible again.)
-  logger lv mg = asks sparCtxLogger >>= \lg -> Spar $ Log.log lg (toLevel lv) mg'
+  logger (toLevel -> lv) mg = do
+    lg <- asks sparCtxLogger
+    reqid <- asks sparCtxRequestId
+    let fields, mg' :: Log.Msg -> Log.Msg
+        fields = Log.field "request" (unRequestId reqid)
+        mg'    = Log.msg . condense $ flatten <$> mg
+    Spar . Log.log lg lv $ fields Log.~~ mg'
     where
-      mg' = Log.msg $ flatten <$> mg
+      condense = dropWhile (== ' ') . reverse . dropWhile (== ' ') . reverse . f
```

Comment:
Looks like a lot of work being done here... is it really necessary to remove all "extra" whitespaces when logging?

Reply:
saml2-web-sso dumps large haskell values into the log in pretty-printed form.  while not strictly necessary, it looks much nicer with `condense`.  please pick what i should do:

- [ ] leave the code as is;
- [ ] remove `condense` entirely;
- [ ] rewrite `condense` to use lazy bytestrings, which should be fast enough given that we're about to write this string to a socket, which should dwarf the overhead for string brushing.

Reply:
Let's try then 3 and perhaps have a 2nd look if the need arises. Particularly since you mention _large_ haskell values.